Paulstar200/Sidehustle-V2-backend
# feat(sign-up): sign up user

### What does this PR do?
- Add Sign up functionality

### Description of Task to be completed?
- Add Sign up controller
- Add user model
- Add validations for sign up fields

### How should this be manually tested?
1. Clone this repo
2. Checkout to the branch `ft-signup-user-170515331`
3. Install all dependencies using `npm install`
4. Start the application using `npm run start:dev`
5. Navigate to Postman to and enter the endpoint `http://localhost:3000/api/v1/auth/register`
6. Create a user using the following fields
```
{
	"firstname": "",
	"lastname": "",
	"username": "",
	"email": "",
	"password": "",
	"confirmPassword": ""
}
```
7. Post and see whether the user is created

### Any background context you want to provide?
- N/A

### What are the relevant pivotal tracker stories?

### Screenshots (if appropriate):
<img width="1440" alt="Screenshot 2020-02-11 at 18 03 19" src="https://user-images.githubusercontent.com/10106044/74248553-ce99ae80-4cf8-11ea-8b0b-c240855dd144.png">

### Questions:
